File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/widgets/BasicBuilder.java
package cz.cvut.fel.matyapav.afandroid.builders.widgets;

import android.content.Context;

import cz.cvut.fel.matyapav.afandroid.components.parts.FieldInfo;
import cz.cvut.fel.matyapav.afandroid.builders.skins.Skin;

/**
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 *
 *@since 1.0.0..
 */
abstract class BasicBuilder implements AbstractWidgetBuilder {

    private Skin skin;
    private FieldInfo properties;
    private Context context;

    BasicBuilder(Context context, Skin skin, FieldInfo properties) {
        this.skin = skin;
        this.properties = properties;
        this.context = context;
    }

    public Skin getSkin() {
        return skin;
    }

    public FieldInfo getProperties() {
        return properties;
    }

    public Context getContext() {
        return context;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/builders/widgets/BasicBuilder.cs
using AFWinPhone.components.parts;
using AFWinPhone.builders.skins;
using Windows.UI.Xaml;

namespace AFWinPhone.builders.widgets
{
    abstract class BasicBuilder : AbstractWidgetBuilder
    {
        private Skin skin;
        private FieldInfo properties;

        public BasicBuilder(Skin skin, FieldInfo properties)
        {
            this.skin = skin;
            this.properties = properties;
        }

        public Skin getSkin()
        {
            return skin;
        }

        public FieldInfo getProperties()
        {
            return properties;
        }

        public abstract FrameworkElement buildFieldView();
        public abstract void setData(AFField field, object value);
        public abstract object getData(AFField field);
    }
}
